versions/development/parsers/antlr4/README.md

```diff
@@ -0,0 +1,87 @@
+for producing parsers of a variety different languages. It is very feature rich and is actively developed and maintained by an active community of contributors.
+
+ANTLR is written in java, but has a variety of other runtime libraries that enable it to be used to generate parses in 8 different languages.
+
+
+### Requirements:
+
+- Java(7+)
+- antlr4
+
+- Python2/3 if you plan on building the python targets
+- Node.js (0.12.7) if you plan on building the JavaScript targets
+- The [runtimes](https://github.com/antlr/antlr4/blob/master/doc/targets.md) for each target you wish to build.
+
+### Resources:
+
+- [ANTLR4 Github](https://github.com/antlr/antlr4)
+- [Mega Tutorial](https://tomassetti.me/antlr-mega-tutorial/)
+- [Grammar Examples](https://github.com/antlr/grammars-v4)
+- [Definitive Reference](https://www.oreilly.com/library/view/the-definitive-antlr/9781941222621/)
+
+
+# Getting Started
+
+To install ANTRL4 locally, please follow the [getting started guide](https://github.com/antlr/antlr4/blob/master/doc/getting-started.md)
+
+# The Grammars
+
+The grammar is divided into two distinct components, a [Lexer](WdlLexer.g4) and a  [Parser](WdlParser.g4). The Lexer is used to define the vocabulary which describes the WDL language in a series of tokens. The parser defines a set of rules which uses the token vocabulary to derive a parse tree of an input script of file. 
+
+There are multiple Lexers within this directory which will result in the exact same tokenization. Each target language will use only one of the lexers, defined
+within the build step for that language. As described [below](#the-base-lexer), the lexers need to handle ambiguous tokens in the case of string and command interpolation. Because of this we need to define actions within the lexers which are more intelligent then a simple pattern match. The actions use predicates instead of native code, however because of the differences between languages, there is no one way to reference the current lexer context within an action that is completely portable. For example in `WdlLexer.g4` all predicates start with `this`, the keyword found in `JavaScript`, `Java`, `C#` etc for accessing the current `object`. However within `Python`, `this` is undefined, and instead `self` must be used as is the case in `WdlLexerPython.g4`.
+
+It is important that the lexers only differ in terms of their actions, and the actual rules stay the same to guarantee that all parsers are equivalent. 
+
+## Parsers
+
+The grammars themselves cannot be used to parse a WDL as is, they first must be converted into a parser in one of the supported target languages. Currently
+Java, Python3, Python2, and JavaScript are supported. For Python and JavaScript you can use the `antlr4` command line tool to generate the required code that is used in the Parsers. For the java target, the easiest way to generate the code is to use the `mvn` plugin defined directly within the `pom.xml`. 
```

Comment:
Let's not merge a PR that combines "python2" and "supported" in one sentence. Python2 will be deprecated in a few days, but it still has a user base. It's much easier to not ship python2 support now than to get rid of it later.